dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/oracle/OracleConnectorExceptions.java

```diff
@@ -16,12 +16,19 @@
  */
 package com.google.edwmigration.dumper.application.dumper.connector.oracle;
 
+import com.google.edwmigration.dumper.application.dumper.MetadataDumperUsageException;
+
 class OracleConnectorExceptions {
 
   static IllegalArgumentException invalidSqlSourcePath(String path) {
     String message = String.format("Invalid path for SQL source file: %s", path);
     return new IllegalArgumentException(message);
   }
 
+  static MetadataDumperUsageException mustProvideServiceOrSid() {
```

Comment:
Optional: I think the `OracleConnectorExceptions` class introduces unnecessary abstraction that causes too much indirection. Unless there are some other requirements that I'm not aware of, by default I would create the exception where it is thrown, so that no other class needs to be consulted when reading the code.

My main argument is simplicity - not to overcomplicate things too much.